bundles/org.eclipse.jface/src/org/eclipse/jface/action/MenuManager.java

```diff
@@ -740,28 +740,17 @@ protected void doItemFill(IContributionItem ci, int index) {
 	protected void update(boolean force, boolean recursive) {
 		if (isDirty() || force) {
 			if (menuExist()) {
-				// clean contains all active items without double separators
 				IContributionItem[] items = getItems();
+
+				// clean contains all active items
 				List<IContributionItem> clean = new ArrayList<>(items.length);
```

Comment:
at "cleaning up" the double separators is not done later: is "clean" still a good name?

Reply:
To be honest, I think "clean" was never a good name.
But after the change, the not visible items are still "cleaned" from the list